### ISSUE TITLE
ContactAnalysis1 issue

### ISSUE THREAD

**Issue by euhruska** [MDAnalysis/mdanalysis#624]
I have some issues with MDAnalysis.analysis.contacts.ContactAnalysis1

First, it refused to work with a universe formed from a list of trajectories. Exactly the same commands with just one trajectory in the universe worked.
The error message for the list of trajectories:

```
>>> CA1 = MDAnalysis.analysis.contacts.ContactAnalysis1(u, selection=('all'), refgroup=(ref1) )
>>> CA1.run()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/MDAnalysis-0.12.1.dev0-py2.7-macosx-10.10-x86_64.egg/MDAnalysis/analysis/contacts.py", line 621, in run
    for ts in self.universe.trajectory[start_frame:end_frame:step_value]:
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/MDAnalysis-0.12.1.dev0-py2.7-macosx-10.10-x86_64.egg/MDAnalysis/coordinates/base.py", line 1008, in __getitem__
    frame.start, frame.stop, frame.step)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/MDAnalysis-0.12.1.dev0-py2.7-macosx-10.10-x86_64.egg/MDAnalysis/coordinates/base.py", line 1055, in _check_slice_indices
    raise TypeError("Slice indices are not integers")
TypeError: Slice indices are not integers
```

Second, for that one trajectory which has 4000 frames, I got just 3999 q-values.
I'm not sure, why there is one frame missing and I have no idea if the first or the last frame is missing. Is this intended behavior?

Third, is it possible to calculate the q-value in the following shape?
for ts in u.trajectory:
  ...calculate q-value

**Comment by orbeckst** [comment]
For (1): Can you reproduce the problem with "chained" trajectories using the test datafiles, e.g.
```python
import MDAnalysis as mda; from MDAnalysis.tests.datafiles import PDF, DCD
u = mda.Universe(PSF, [DCD, DCD])
```

For (2): Please provide an example with the bundled PSF, DCD that we can just copy & paste :-).

**Comment by euhruska** [comment]
I believe (1) solved itself after I reinstalled MDAnalysis, but (2) ist still here:
```
import MDAnalysis
from MDAnalysis.tests.datafiles import PSF,DCD  
import MDAnalysis.analysis.contacts 
u = MDAnalysis.Universe(PSF,DCD) 
sel_basic = "(resname ARG or resname LYS) and (name NH* or name NZ)"
sel_acidic = "(resname ASP or resname GLU) and (name OE* or name OD*)"

# reference groups (first frame of the trajectory, but you could also use a separate PDB, eg crystal structure)
acidic = u.select_atoms(sel_acidic)
basic = u.select_atoms(sel_basic)

# set up analysis of native contacts ("salt bridges"); salt bridges have a distance <6 A
CA1 = MDAnalysis.analysis.contacts.ContactAnalysis1(u, selection=(sel_acidic, sel_basic), refgroup=(acidic, basic), radius=6.0, outfile="qsalt.dat")

# iterate through trajectory and perform analysis of "native contacts" q
# (force=True ignores any previous results, force=True is useful when testing)
CA1.run(force=True)
u.trajectory.n_frames
CA1.timeseries.shape
```
n_frames and the length of the timeseries should be the same, but I get a difference of 1.

**Comment by tylerjereddy** [comment]
The problem is here:

```python
  def run(self, store=True, force=False, start_frame=1, end_frame=None, step_value=1):
```
at this location in the codebase / commit hash:
https://github.com/MDAnalysis/mdanalysis/blob/ae755d8da449d73d8f3ecff4d8015affa742d236/package/MDAnalysis/analysis/contacts.py#L587

As a temporary fix, you could set start_frame = 0 to capture the first frame. Currently, indexing starts at the second element of the trajectory by default because `start_frame` is used as the starting slice index for the trajectory object.

I'll work on a (hopefully simple) PR for this.

**Comment by orbeckst** [comment]
Thanks Tyler. Seems that this was left over from when we moved to 0-based frame indexing. That should be an easy fix for 0.13.1.

I also suggest to use the more canonical keywords start, stop, step. Can you add those and deprecate start_frame and end_frame? 

On 19 Jan, 2016, at 08:39, Tyler Reddy wrote:

> The problem is here:
> 
>   def run(self, store=True, force=False, start_frame=1, end_frame=None, step_value=1):
> at this location in the codebase / commit hash:
> https://github.com/MDAnalysis/mdanalysis/blob/ae755d8da449d73d8f3ecff4d8015affa742d236/package/MDAnalysis/analysis/contacts.py#L587
> 
> As a temporary fix, you could set start_frame = 0 to capture the first frame. Currently, indexing starts at the second element of the trajectory by default because start_frame is used as the starting slice index for the trajectory object.
> 
> I'll work on a (hopefully simple) PR for this.
> 
> �
> Reply to this email directly or view it on GitHub.
> 

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com
